packages/application-extension/src/index.tsx

```diff
@@ -1502,6 +1518,16 @@ namespace Private {
         }
       }
     });
+
+    // Handle disabled status.
+    setDisabled(settings);
+    commands.addCommand(CommandIDs.toggleContextMenu, {
+      label: trans.__('Enable Context (right-click) Menu'),
```

Comment:
I'm not sure if this should be in the main Settings menu - is this something that users would toggle frequently enough to justify it?

I would say it should not, but if it does need to be in there, I would suggest to make the label shorter, also because depending on a device it will not be a right-click, see https://github.com/jupyterlab/jupyterlab/issues/14682#issuecomment-2351836843.

In any case, I think it could be added to the command palette.

```suggestion
      label: trans.__('Enable Context Menu'),
      caption: trans.__('Enable context (right-click) menu'),
```

Reply:
"Context menu" is a term of art that menu users might not know, that's why I added `right-click`, but I do agree it's clunky and I don't have strong feelings about it.

The reason I added it to the `Settings` menu is because it's not at all discoverable as a setting otherwise, because the context menu plugin is specifically omitted in the visual settings editor.

But actually maybe that's fine because this PR can add the feature and another follow-on PR can make it more discoverable

Reply:
I will remove the command from the `Settings` menu and maybe add it to the command palette.

I will also remove the `right-click` language and another PR can deal with verbiage and discoverability if we find that there's a problem.

Reply:
> because the context menu plugin is specifically omitted in the visual settings editor

I think this is the problem we should address. I think it should not be omitted. I think we should discuss it separately, but my memory is that:
- setting editor UI was introduced as a huge list rendered all at once with accordion panels
- it was slow, so some plugins with lot's of settings were excluded which improved the performance
- it was then rewritten to only show one plugin at a time
- the plugins which were previously omitted for performance reasons were not re-enabled

So maybe these should be excluded from search as when searching "c" too much would render, but I do not think we should hide them.

Reply:
I opened https://github.com/jupyterlab/jupyterlab/issues/17362 to track it.